core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -944,50 +951,44 @@ private void removeStaleData(final Set<Integer> removedSegments) throws Interrup
       // gather all keys from cache store that belong to the segments that are being removed/moved to L1
       if (!removedSegments.isEmpty()) {
          try {
-            KeyFilter filter = new KeyFilter() {
-               @Override
-               public boolean accept(Object key) {
-                  if (dataContainer.containsKey(key))
-                     return false;
-                  int keySegment = getSegment(key);
-                  return (removedSegments.contains(keySegment));
-               }
+            KeyFilter filter = key -> {
+               if (dataContainer.containsKey(key))
+                  return false;
+               int keySegment = getSegment(key);
+               return (removedSegments.contains(keySegment));
             };
-            persistenceManager.processOnAllStores(filter, new AdvancedCacheLoader.CacheLoaderTask() {
-               @Override
-               public void processEntry(MarshalledEntry marshalledEntry, AdvancedCacheLoader.TaskContext taskContext) throws InterruptedException {
-                  keysToRemove.add(marshalledEntry.getKey());
-               }
-            }, false, false, PRIVATE);
+            persistenceManager.processOnAllStores(filter,
+                  (marshalledEntry, taskContext) -> keysToRemove.add(marshalledEntry.getKey()), false, false, PRIVATE);
          } catch (CacheException e) {
             log.failedLoadingKeysFromCacheStore(e);
          }
       }
 
       if (!keysToRemove.isEmpty()) {
          try {
-            InvalidateCommand invalidateCmd = commandsFactory.buildInvalidateCommand(EnumUtil.bitSetOf(CACHE_MODE_LOCAL, SKIP_LOCKING), keysToRemove.toArray());
+            InvalidateCommand invalidateCmd = commandsFactory
+                  .buildInvalidateCommand(EnumUtil.bitSetOf(CACHE_MODE_LOCAL, SKIP_LOCKING), keysToRemove.toArray());
             InvocationContext ctx = icf.createNonTxInvocationContext();
             ctx.setLockOwner(invalidateCmd.getKeyLockOwner());
             interceptorChain.invoke(ctx, invalidateCmd);
 
-            if (trace) log.tracef("Removed %d keys, data container now has %d keys", keysToRemove.size(), dataContainer.size());
+            if (trace)
```

Comment:
@slaskawi You can try, but I doubt it -- the difference is always going to be visible, especially for fast operations like replicated mode reads.